packages/react/package.json

```diff
@@ -42,6 +42,7 @@
     "@babel/preset-stage-0": "^7.8.3",
     "@babel/preset-typescript": "^7.9.0",
     "@babel/register": "^7.9.0",
+    "@radix-ui/react-polymorphic": "^0.0.14",
```

Comment:
I tried out several libraries and evaluated creating the prop types internally. Other libraries were either unreliable or required code changes to the runtime, which I was trying to avoid. If anything, if we decide not to use this library directly, we should consider copying the code in and citing the source.

Reply:
Do we really need a library for this? It seems [pretty simple to do on our own](https://github.com/segmentio/ui-box/blob/master/src/types/box-types.ts).

Reply:
I don't mind pulling this in. I have not evaluated segmentio's implementation. The end implementation is not overly complicated. However, getting there is pretty finicky. Other implementations I looked at (libraries, blog posts, etc.) either had shortcomings when it came to refs, allowed you to specify `as` but did not enforce the prop types, or slowed down Typescript a ton.

I went ahead and brought polymorphic in and referenced the source material